docs/source/contributing.rst

```diff
@@ -1,3 +1,88 @@
-Contributing
-============
+How to contribute to Orchestra
+==============================
 
+So you want to get involved in developing Orchestra. Great! We're excited to
+have your support. This document lays down a few ground rules to help the whole
+process run smoothly.
+
+Getting involved
+################
+
+First, if you find a bug in the code or documentation, check out
+our `open issues <https://github.com/unlimitedlabs/orchestra/issues>`_ and
+`pull requests <https://github.com/unlimitedlabs/orchestra/pulls>`_ to see if
+we're already aware of the problem. Also feel free to reach out to us on
+`gitter <https://gitter.im/unlimitedlabs/orchestra>`_ to answer questions at
+any time.
+
+If you've uncovered something new, please `create an issue
+<https://github.com/unlimitedlabs/orchestra/issues>`_ describing the problem.
+Once you've written code that fixes the issue, `create a pull request
+<https://help.github.com/articles/creating-a-pull-request/>`_ (PR) so it's easy
+for us to incorporate your changes.
+
+Development Workflow
+####################
+
+Github provides a nice `overview on how to create a pull request
+<https://help.github.com/articles/creating-a-pull-request/>`_.
+
+Some general rules to follow:
+
+* Do your work in a `fork <https://help.github.com/articles/fork-a-repo/>`_
+  of the Orchestra repo.
+
+* Create a branch for each feature/bug in Orchestra that you're working on.
+  These branches are often called "feature" or "topic" branches.
+
+* Use your feature branch in the pull request. Any changes that you push to
+  your feature branch will automatically be shown in the pull request.
+
+* Keep your pull requests as small as possible. Large pull requests are hard to
+  review. Try to break up your changes into self-contained and incremental pull
+  requests, if need be, and reference dependent pull requests, e.g. "This pull
+  request builds on request #92. Please review #92 first."
+
+* Include unit tests with your pull request. We love tests and use `CircleCI
+  <https://circleci.com/>`_ to check every pull request and commit.
+  Check out our tests in ``orchestra/tests`` to see examples of how to write
+  unit tests. Before submitting a PR, make sure that running ``make test`` from
+  the root directory of the repository succeeds.
+
+* Once you submit a PR, you'll get feedback on your code, sometimes asking for
+  a round or two of changes before your PR is accepted. After each round of
+  comments, make changes accordingly, then squash all changes for that round
+  into a single commit with a name like 'changes round 0'.
+
+* After your PR is accepted, your should squash all of your changes into a
```

Comment:
any quick instructions for this?  something like


         git checkout master
         git pull
         git checkout your-branch
         git rebase master
         git log
         <find the commit hash that happened immediately before your first commit (the most recent commit hash that isn't your commit)>
         git reset --soft $THAT_COMMIT_HASH$
         git commit -am "A commit message that summarizes your change in a single line"
         git push -f origin your-branch